scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,172 @@
 package scenelib.annotations.el;
 
+import org.objectweb.asm.Label;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
 import java.util.Objects;
 
+import org.checkerframework.checker.nullness.qual.Nullable;
+
 /**
- * A {@link LocalLocation} holds location information for a local
- * variable: slot index, scope start, and scope length.
+ * A {@link LocalLocation} holds information about a local variable.
+ * A variable may have multiple lifetimes.
+ * We store this information the same way ASM does, as 3 parallel arrays.
  */
 public final class LocalLocation {
     /**
-     * The slot index of the local variable.
+     * The starts of the lifetimes of the element being visited.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
      */
-    public final int index;
+    public final Label[] start;
 
     /**
-     * The start of the local variable's scope (or live range), as an offset
-     * from the beginning of the method code in bytes.
+     * The ends of the lifetimes of the element being visited.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
      */
-    public final int scopeStart;
+    public final Label[] end;
 
     /**
-     * The length of the local variable's scope (or live range), in bytes.
+     * The indices of the element being visited in the classfile.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
      */
-    public final int scopeLength;
+    public final int[] index;
 
-    public final String varName;
-    public final int varIndex;
+    /**
+     * The name of the local variable being visited.
+     *
+     * This is not part of the abstract state of the LocalLocation:
+     * it is not read by equals(), hashCode(), or toString().
+     */
+    public final @Nullable String variableName;
 
     /**
-     * Constructs a new {@link LocalLocation}; the arguments are assigned to
-     * the fields of the same names.
+     * Construct a new LocalLocation.  This constructor does not assign meaningful
+     * values to start or end.  Thus, the getScopeStart and getScopeLenth methods must not
+     * be used on the result.
+     *
+     * @param variableName the name of the local variable
+     * @param index the offset of the variable in the stack frame
      */
-    public LocalLocation(int index, int scopeStart, int scopeLength) {
+    public LocalLocation(String variableName, int index) {
```

Comment:
The order of arguments in this constructor is inconsistent with others, in that the variable name comes first rather than last.